app/Judite/Models/Student.php

```diff
@@ -112,11 +128,23 @@ public function isEnrolledInCourse(Course $course): bool
      *
      * @param \App\Judite\Models\Course $course
      *
+     * @throws \App\Exceptions\StudentIsNotEnrolledInCourseException|\App\Exceptions\EnrollmentCannotBeDeleted
+     *
      * @return bool
      */
     public function unenroll(Course $course): bool
     {
-        return $this->enrollments()->where('course_id', $course->id)->delete();
+        $enrollment = $this->getEnrollmentByCourse($course);
+
+        if (is_null($enrollment)) {
+            throw new StudentIsNotEnrolledInCourseException($course);
+        }
+
+        if (is_null($enrollment->shift)) {
+            return $this->enrollments()->whereCourseId($course->id)->delete();
```

Comment:
This line can be replaced with `$enrollment->delete()`.

Additionally, swap the `return` with the `throw`.
The `else` block is unnecessary.